### PR TITLE
feat: add unit tests for manager handlers

### PR DESCRIPTION
**Pull request by BruceAko**
## Description

Add unit tests for manager handlers.

## Related Issue

fix https://github.com/dragonflyoss/Dragonfly2/issues/2705

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by BruceAko** [comment]
@gaius-qi So far I've written unit tests for `application.go`, could you please review this part of the code so that I can make sure I'm writing the unit tests in the right way? Thank you very much :)

**Comment by gaius-qi** [comment]
> @gaius-qi So far I've written unit tests for `application.go`, could you please review this part of the code so that I can make sure I'm writing the unit tests in the right way? Thank you very much :)

@BruceAko I think it is good！

**Comment by gaius-qi** [comment]
@BruceAko Please fix CI.

**Comment by gaius-qi** [comment]
@BruceAko Have you finished PR?

**Comment by BruceAko** [comment]
@gaius-qi I have finished PR now. Please take a look.

File: manager/handlers/application_test.go [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+		{
+			name: "success",
+			req:  httptest.NewRequest("POST", "/api/v1/applications", strings.NewReader(mockApplicationReqBody)),
+			mock: func(ms *mocks.MockServiceMockRecorder) {
+				ms.CreateApplication(gomock.Any(), gomock.Eq(mockCreateApplicationRequest)).Return(mockApplicationModel, nil).Times(1)
+			},
+			expect: func(t *testing.T, w *httptest.ResponseRecorder) {
+				assert := assert.New(t)
+				assert.Equal(http.StatusOK, w.Code)
+				assert.Equal(w.Body.String(), mockApplicationResponseBody)
```
Comment (gaius-qi) [inline manager/handlers/application_test.go:138]: Unmarshal `w.Body.String()` to struct, remove `mockApplicationReqBody`.

```golang
application := models.Application{}
if err := json.Unmarshal(...) {
...
}

assert.Equal(resp, models.Application{...})
```

Please modify the same type of cases.

File: manager/handlers/application_test.go [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+func TestHandlers_CreateApplication(t *testing.T) {
+	tests := []struct {
+		name   string
+		req    *http.Request
+		mock   func(ms *mocks.MockServiceMockRecorder)
+		expect func(t *testing.T, w *httptest.ResponseRecorder)
+	}{
+		{
+			name: "unprocessable entity",
+			req:  httptest.NewRequest("POST", "/api/v1/applications", nil),
```
Comment (gaius-qi) [inline manager/handlers/application_test.go:122]: Replace `POST` with `http.MethodPost`.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Comment by gaius-qi** [comment]
@BruceAko Can you add unit tests for `handlers.go` in next PR? Thx

**Comment by BruceAko** [comment]
> @BruceAko Can you add unit tests for `�handlers.go` in next PR? Thx

@gaius-qi Ok, sure. And I fixed the changes you raised in two new commits, PTAL.

**Pull request merged**